x-pack/plugins/apm/server/routes/source_maps.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Boom from '@hapi/boom';
+import * as t from 'io-ts';
+import {
+  createApmArtifact,
+  deleteApmArtifact,
+  listArtifacts,
+  updateSourceMapsOnFleetPolicies,
+} from '../lib/fleet/source_maps';
+import { createApmServerRoute } from './create_apm_server_route';
+import { createApmServerRouteRepository } from './create_apm_server_route_repository';
+
+export const sourceMapRt = t.intersection([
+  t.type({
+    version: t.number,
+    sources: t.array(t.string),
+    mappings: t.string,
+  }),
+  t.partial({
+    names: t.array(t.string),
+    file: t.string,
+    sourceRoot: t.string,
+    sourcesContent: t.array(t.string),
+  }),
+]);
+
+const listSourceMapRoute = createApmServerRoute({
+  endpoint: 'GET /api/apm/sourcemaps',
+  options: { tags: ['access:apm'] },
+  handler: async ({ plugins, logger }) => {
+    try {
+      const fleetPluginStart = await plugins.fleet?.start();
+      if (fleetPluginStart) {
+        const artifacts = await listArtifacts({ fleetPluginStart });
+        return { artifacts };
+      }
+    } catch (e) {
+      throw Boom.internal(
+        'Something went wrong while fetching artifacts source maps',
+        e
+      );
+    }
+  },
+});
+
+const uploadSourceMapRoute = createApmServerRoute({
+  endpoint: 'POST /api/apm/sourcemaps/{serviceName}/{serviceVersion}',
+  options: { tags: ['access:apm'] },
```

Comment:
should the upload and delete routes use `tags: ['access:apm', 'access:apm_write']`?